pkg/implementations/spec/spec.go

```diff
@@ -0,0 +1,285 @@
+/*
+Copyright © 2022 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spec
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/kairos-io/kairos-agent/v2/pkg/constants"
+	"github.com/kairos-io/kairos-sdk/ghw"
+	"github.com/kairos-io/kairos-sdk/types/images"
+	"github.com/kairos-io/kairos-sdk/types/partitions"
+)
+
+// InstallSpec struct represents all the installation action details
+type InstallSpec struct {
```

Comment:
Wait...didn't we move these to the kairos-sdk? The title of this PR also says that the kairos-sdk will provide "structs".

Reply:
Yeah but not all of them are relevant or generic enough to be reused. Something like the Kairos ask or w generic image extractor makes sense, but the spec for the  agent seems very narrow in their use and we don't expect it to be needed by anything else